### PR TITLE
dust: adding new package

### PR DESCRIPTION
**Pull request by fangohr**
dust = du + rust. Like du, but more intuitive.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dust/package.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    maintainers = ["fangohr"]
+
+    version(
+        "0.7.5",
+        sha256="f892aaf7a0a7852e12d01b2ced6c2484fb6dc5fe7562abdf0c44a2d08aa52618",
+    )
+
+    depends_on("rust")
+
+    sanity_check_is_file = [join_path("bin", "dust")]
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/dust/package.py:24]: I don't see this used anywhere, is it needed for something? If not, can you remove it?

**Review by scheibelp — COMMENTED** [review]
I have one question, but otherwise this looks good.

File: var/spack/repos/builtin/packages/dust/package.py (hunk 1 above)
Comment (fangohr) [reply var/spack/repos/builtin/packages/dust/package.py:24]: The line checks if the file `[installation-prefix]/bin/dust` is created when the package is installed. If it isn't, it raises an exception. 

The use of these sanity checks is mentioned in the packaging guide (https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-sanity-checks).

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**